wasm/internal/readpos/readpos.go

```diff
@@ -0,0 +1,22 @@
+package readpos
+
+import (
+	"io"
+)
+
+type ReadPos struct {
```

Comment:
please document `ReadPos`.
e.g.

```go
// ReadPos is an io.Reader that counts the number of bytes read.
type ReadPos struct {
...
```

there is a bit of stuttering: `var r readpos.ReadPos`.
perhaps:
```go
type Reader struct {
  R io.Reader
  N uint64 // N is the number of bytes read from R
}
```
also: is `uint64` really needed? wouldn't `int64` be more natural and less error-prone (to wrap around) when computing differences/subtractions?
(the `io.Seeker` uses `int64` for e.g.)